src/borg/helpers/datastruct.py

```diff
@@ -49,3 +49,58 @@ def get(self, size=None, init=False):
         if size is not None:
             self.resize(size, init)
         return self.buffer
+
+
+class EfficientBytesQueue:
```

Comment:
with a little change, i guess this could also be generalized to have any sequence type as elements (not just bytes).

not likely to be used by borg though, so guess this is optional...